### PR TITLE
Delays initialization until Main.panel.statusArea.quickSettings._system is available

### PR DESCRIPTION
**Pull request by lobothiago**
This fixes https://github.com/arelange/gnome-shell-extension-hibernate-status/issues/110, although I'm not sure the fix is appropriate given gnome-shell extension best practices :)

### CODE REVIEW

File: extension.js [hunk 1]
```diff
@@ -363,59 +363,66 @@ export default class MyExtension extends Extension {
...
-
-        this._menuOpenStateChangedId = this.systemMenu._systemItem.menu.connect(
-            'open-state-changed',
-            (menu, open) => {
-                if (!open) return;
-                this._updateDefaults();
-                this._updateHaveHibernate();
-                this._updateHaveHybridSleep();
-                this._updateHaveSuspendThenHibernate();
+        var self = this;
```
Comment (p91paul) [inline extension.js:366]: the => javascript operator AFAIK captures this, making the usual var self = this unnecessary. Also, people nowadays discourage using var in favour of let or const.

**Review by p91paul — CHANGES_REQUESTED** [review]
I think it would be better to know why in some systems the extension is enabled too early, rather than polling, but I cannot reproduce. I leave to @slaclau to decide what to do with this PR

**Comment by p91paul** [comment]
https://github.com/UbayGD/reboottouefi/blob/main/src/extension.js seems to be using a different and cleaner approach in the enable method

File: extension.js (hunk 1 above)
Comment (lobothiago) [reply extension.js]: Thanks for the feedback! I have updated the polling mechanism as per https://github.com/UbayGD/reboottouefi/blob/main/src/extension.js and as recommended in https://discourse.gnome.org/t/main-panel-statusarea-quicksettings-system-is-undefined/16827. Working on my machine.

**Review by slaclau — COMMENTED** [review]
I'm happy with these changes and they appear to follow best practice except for removing sourceId on disable.

**Pull request merged**